### ISSUE TITLE
[Cleanup] Orphaned SamplingParams.output_text_buffer_length and stale spec-decode startup warning after V0 removal

### ISSUE THREAD

**Issue by AdaAibaby** [vllm-project/vllm#50217]
## Summary

Two dead items were left behind after the V0 deprecation:

### 1. `SamplingParams.output_text_buffer_length` is computed but never consumed

**File:** `vllm/sampling_params.py` lines 316 and 492–495

The field was introduced in #3672 to track the number of characters to hold back during streaming stop-string evaluation. It was consumed by the V0 `from_seq_group` path, which was removed in #25330.

The V1 detokenizer independently recomputes the same value as a local variable (`stop_buffer_length`) in `vllm/v1/engine/detokenizer.py:88`:

```python
self.stop_buffer_length = max(len(s) for s in self.stop) - 1
```

The `SamplingParams` field is now orphaned: `__post_init__` computes it on every request that has stop strings, it is serialised over IPC as part of the struct, but no code path ever reads `sampling_params.output_text_buffer_length`.

### 2. Stale `logger.warning` in `build_logitsprocs` contradicts current behaviour

**File:** `vllm/v1/sample/logits_processor/__init__.py` lines 205–207

```python
logger.warning(
    "min_p and logit_bias parameters won't work with speculative decoding."
)
```

This warning was written when `min_p`/`logit_bias` with spec decode caused a **silent drop** (the logits processors were simply not built). Since then, `_validate_spec_decode` (`vllm/sampling_params.py:887`) was added to raise a hard `VLLMValidationError` per request instead.

The startup warning now describes behaviour that no longer exists. An operator reading it concludes "requests using `min_p` or `logit_bias` will silently produce output with degraded sampling" — the actual behaviour is "those requests are rejected with HTTP 400". The warning fires every server startup with spec decode enabled, creating noise.

## Verification

```bash
# Confirm output_text_buffer_length has no consumers:
grep -rn "output_text_buffer_length" vllm/ --include="*.py"
# Returns only the two lines being removed.

# Confirm V1 detokenizer has its own local computation:
grep -n "stop_buffer_length" vllm/v1/engine/detokenizer.py

# Confirm _validate_spec_decode raises per-request:
grep -n "_validate_spec_decode\|VLLMValidationError" vllm/sampling_params.py | grep -A3 "spec_decode"
```

## No existing PR

Checked with:
```
gh pr list --repo vllm-project/vllm --state open --search "output_text_buffer_length"
gh pr list --repo vllm-project/vllm --state open --search "spec decode warning logit_bias"
```
No match found.

*AI assistance was used to identify these findings. All lines have been reviewed manually.*
